src/main/java/sg/ncl/MainController.java

```diff
@@ -950,14 +953,14 @@ private void registerUserToDeter(JSONObject mainObject) throws
                     case USERNAME_ALREADY_EXISTS_EXCEPTION:
                         // throw from user service
                     {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Extract this nested code block into a method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1199)